buzzmoon-frontend/buzzmoon-vite/src/Components/Results/Results.jsx

```diff
@@ -6,10 +6,65 @@ import './Results.css';
 //placeholder results page
 export default function Results(props) {
   const {gameID} = useParams();
+  const [gameData, setGameData] = React.useState();
+  const [resultsCat, setResultsCat] = React.useState("normal");
+  const [table, setTable] = React.useState([]);
+
+  React.useEffect(() => {
+    const updateGameData = async () => {
+      const fetchedGameData = await BackendActor.getGameMetadata(gameID);
+      setGameData(fetchedGameData);
+      
+
+      // here we should validate if the user has played the game, or if they should be kicked from results page
+
+      const playerResults = await BackendActor.fetchGameResults(gameID);
+      const table = BackendActor.resultsToStandardTable(playerResults);
+      setTable(table);    
+    }
+
+    updateGameData();
+  }, []);
+
+  
+  if (! gameData) {
+    return (
+      <div className='results'>
+        <i className="fa-solid fa-spinner fa-spin"></i>
```

Comment:
Might be worthwhile checking out React.Suspense. Allows you to “wait” for some code to load and declaratively specify a loading state (i.e spinner) while waiting